src/main/kotlin/gg/flyte/christmas/minigame/games/Spleef.kt

```diff
@@ -10,155 +11,214 @@ import gg.flyte.christmas.util.eventController
 import gg.flyte.christmas.util.formatInventory
 import gg.flyte.christmas.util.style
 import gg.flyte.twilight.event.event
+import gg.flyte.twilight.extension.getNearestPlayer
 import gg.flyte.twilight.extension.hidePlayer
 import gg.flyte.twilight.extension.playSound
 import gg.flyte.twilight.extension.showPlayer
 import gg.flyte.twilight.scheduler.TwilightRunnable
 import gg.flyte.twilight.scheduler.delay
 import gg.flyte.twilight.scheduler.repeatingTask
+import net.kyori.adventure.bossbar.BossBar
 import net.kyori.adventure.text.Component
-import org.bukkit.Bukkit
-import org.bukkit.GameMode
-import org.bukkit.Material
-import org.bukkit.Sound
+import net.minecraft.server.level.ServerLevel
+import net.minecraft.sounds.SoundEvents
+import net.minecraft.world.entity.EntityType
+import net.minecraft.world.entity.LivingEntity
+import net.minecraft.world.entity.ai.goal.RangedAttackGoal
+import net.minecraft.world.entity.animal.SnowGolem
+import net.minecraft.world.entity.projectile.Projectile
+import net.minecraft.world.item.Items
+import net.minecraft.world.level.Level
+import org.bukkit.*
+import org.bukkit.attribute.Attribute
+import org.bukkit.block.Block
 import org.bukkit.block.data.type.Snow
-import org.bukkit.entity.ItemDisplay
-import org.bukkit.entity.Player
-import org.bukkit.entity.Snowball
+import org.bukkit.entity.*
 import org.bukkit.event.block.BlockBreakEvent
+import org.bukkit.event.entity.CreatureSpawnEvent
 import org.bukkit.event.entity.ProjectileHitEvent
-import org.bukkit.event.player.PlayerMoveEvent
+import org.bukkit.event.entity.ProjectileLaunchEvent
+import org.bukkit.event.player.PlayerDropItemEvent
 import org.bukkit.event.player.PlayerToggleFlightEvent
 import org.bukkit.inventory.ItemStack
 import org.bukkit.potion.PotionEffect
 import org.bukkit.potion.PotionEffectType
 import org.bukkit.util.Vector
+import java.util.*
+import kotlin.math.sqrt
 
 class Spleef : EventMiniGame(GameConfig.SPLEEF) {
     private var overviewTasks = mutableListOf<TwilightRunnable>()
+
     private var floorLevelBlocks: List<MapSinglePoint> = listOf(
         Util.fillArena(110, Material.SNOW_BLOCK),
         Util.fillArena(98, Material.SNOW_BLOCK),
-        Util.fillArena(86, Material.SNOW_BLOCK),
-    ).flatMap { it }
-    private var seconds = 0
-    private var doubleJumps = mutableMapOf<Player, Int>()
+        Util.fillArena(86, Material.SNOW_BLOCK)
+    ).flatten()
+
+    private var secondsElapsed = 0
+
+    private var doubleJumps = mutableMapOf<UUID, Int>()
+
+    private var unlimitedJumps = false
+    private var remainingUnlimitedJumpTicks = 0
```

Comment:
I have to use both variables because I need to know the moment when `remainingUnlimitedJumpTicks` is set from 0 to something else. Otherwise I would end up spamming `it.allowFlight = true` and `it.showBossBar()`